### PR TITLE
fix: use dynamic components in `root.svelte` instead of `svelte:component` for svelte 5

### LINKED ISSUE #12573: With `compilerOptions: { runes: true }`, `root.svelte` file generate `` `<svelte:component>` is deprecated in runes mode — components are dynamic by default``
### Describe the bug

With `compilerOptions: { runes: true }`, `root.svelte` file generate `` `<svelte:component>` is deprecated in runes mode — components are dynamic by default``

I'm not sure if this is a bug that SvelteKit, Svelte, or @svelte/adapter-static needs to fix. Hope you guys can transfer it to the correct one if needed.

### Reproduction

Just create a new sveltekit project and enable runes only mode by adding `compilerOptions: { runes: true }` to the `svelte.config.js` file.

### Logs

```Shell
5:55:38 PM [vite-plugin-svelte] .svelte-kit/generated/root.svelte:45:1 `<svelte:component>` is deprecated in runes mode — components are dynamic by default
43: {#if constructors[1]}
44:   <!-- svelte-ignore binding_property_non_reactive -->
45:   <svelte:component this={constructors[0]} bind:this={components[0]} data={data_0}>
                         ^
46:     <!-- svelte-ignore binding_property_non_reactive -->
47:     <svelte:component this={constructors[1]} bind:this={components[1]} data={data_1} {form} />
5:55:38 PM [vite-plugin-svelte] .svelte-kit/generated/root.svelte:47:2 `<svelte:component>` is deprecated in runes mode — components are dynamic by default
45:   <svelte:component this={constructors[0]} bind:this={components[0]} data={data_0}>
46:     <!-- svelte-ignore binding_property_non_reactive -->
47:     <svelte:component this={constructors[1]} bind:this={components[1]} data={data_1} {form} />
                                                                                                  ^
48:   </svelte:component>
49: {:else}
5:55:38 PM [vite-plugin-svelte] .svelte-kit/generated/root.svelte:51:1 `<svelte:component>` is deprecated in runes mode — components are dynamic by default
49: {:else}
50:   <!-- svelte-ignore binding_property_non_reactive -->
51:   <svelte:component this={constructors[0]} bind:this={components[0]} data={data_0} {form} />
                                                                                                ^
52: {/if}
53:
```

### System Info

```Shell
System:
    OS: Linux 6.5 KDE neon 6.0 6.0
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
    Memory: 8.43 GB / 15.40 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 22.6.0 - /run/user/1000/fnm_multishells/33079_1723484947088/bin/node
    npm: 10.8.2 - /run/user/1000/fnm_multishells/33079_1723484947088/bin/npm
    bun: 1.1.16 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 127.1.68.137
  npmPackages:
    @sveltejs/adapter-static: ^3.0.3 => 3.0.3 
    @sveltejs/kit: ^2.5.21 => 2.5.21 
    @sveltejs/vite-plugin-svelte: ^3.1.1 => 3.1.1 
    svelte: ^5.0.0-next.218 => 5.0.0-next.218 
    vite: ^5.4.0 => 5.4.0
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by paoloricciuti**
Closes #12573

Again i'm not really sure if and where i should add a test for this. Also for the moment i did not rename `constructors` to something else in svelte 5 because there was still the possibility to share some code but i'm willing to completely split the two branches if we feel like.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: packages/kit/src/core/sync/write_root.js [hunk 1]
```diff
@@ -22,20 +22,39 @@ export function write_root(manifest_data, output) {
 	let l = max_depth;
 
 	let pyramid = dedent`
-	${isSvelte5Plus() ? '<!-- svelte-ignore binding_property_non_reactive -->' : ''}
-	<svelte:component this={constructors[${l}]} bind:this={components[${l}]} data={data_${l}} {form} />
-	`;
+	${
+		isSvelte5Plus()
+			? `<!-- svelte-ignore binding_property_non_reactive -->
+		<Piramid_${l} bind:this={components[${l}]} data={data_${l}} {form} />`
```
Comment (dominikg) [inline packages/kit/src/core/sync/write_root.js:28]: Piramid or Pyramid  ?

Comment (paoloricciuti) [reply packages/kit/src/core/sync/write_root.js:28]: Ugh...will rename

Comment (Conduitry) [reply packages/kit/src/core/sync/write_root.js]: pyramyd

Comment (dominikg) [reply packages/kit/src/core/sync/write_root.js]: :close_snype:

**Review by dummdidumm — APPROVED** [review]

**Pull request merged**